kubernetes client cannot retrieve VersionInfo in native executable

### Describe the bug

Calling KubernetesClient::getKubernetesVersion() fails with a serialization error.

This is blocking as the quarkus operator sdk is fetching the version on bootstrap and fails to start.  see https://github.com/quarkiverse/quarkus-operator-sdk/issues/175.

I can add a `@RegisterForReflection(targets = {VersionInfo.class})` to fix it but it should be handle by the kubernetes client extension.



### Expected behavior

Should deserialize `io.fabric8.kubernetes.client.VersionInfo` properly out of the box.

### Actual behavior

```
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
        at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:314)
        at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:266)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:569)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:527)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:510)
        at io.fabric8.kubernetes.client.dsl.base.OperationSupport.restCall(OperationSupport.java:782)
        at io.fabric8.kubernetes.client.BaseClient.getVersionInfo(BaseClient.java:174)
        at io.fabric8.kubernetes.client.BaseKubernetesClient.getKubernetesVersion(BaseKubernetesClient.java:499)
        at org.acme.GreetingResource.hello(GreetingResource.java:21)
        at java.lang.reflect.Method.invoke(Method.java:566)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
        at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
        at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:408)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:69)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
        ... 17 more
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `io.fabric8.kubernetes.client.VersionInfo` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
 at [Source: (BufferedInputStream); line: 2, column: 3]
        at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1764)
```

### How to Reproduce?

1. Generate a new app with Kubernetes Client extension
2. Modify the GreetingResource with
```
@Path("/hello")
public class GreetingResource {

    @Inject
    KubernetesClient client;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return client.getKubernetesVersion().getPlatform();
    }
}
```

3. `curl /hello`

### Output of `uname -a` or `ver`

Linux NARUTO 5.10.60.1-microsoft-standard-WSL2 #1 SMP Wed Aug 25 23:20:18 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.13" 2021-10-19 OpenJDK Runtime Environment (build 11.0.13+8-Ubuntu-0ubuntu1.20.04) OpenJDK 64-Bit Server VM (build 11.0.13+8-Ubuntu-0ubuntu1.20.04, mixed mode, sharing)

### GraalVM version (if different from Java)

GraalVM 21.3.0 Java 11 CE (Java Version 11.0.13+7-jvmci-21.3-b05)

### Quarkus version or git rev

2.5.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)

### Additional information

command line use to build the native app: `./mvnw package -Pnative -Dquarkus.native.container-build=true`

--- a/extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientProcessor.java
+++ b/extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientProcessor.java
@@ -142,6 +142,8 @@ public class KubernetesClientProcessor {
                 .produce(new ReflectiveClassBuildItem(true, false, "io.fabric8.kubernetes.api.model.IntOrString"));
         reflectiveClasses
                 .produce(new ReflectiveClassBuildItem(true, false, "io.fabric8.kubernetes.internal.KubernetesDeserializer"));
+        reflectiveClasses
+                .produce(new ReflectiveClassBuildItem(true, true, "io.fabric8.kubernetes.client.VersionInfo"));
 
         if (log.isDebugEnabled()) {
             final String watchedClassNames = watchedClasses

--- a/integration-tests/kubernetes-client/src/main/java/io/quarkus/it/kubernetes/client/Version.java
+++ b/integration-tests/kubernetes-client/src/main/java/io/quarkus/it/kubernetes/client/Version.java
@@ -1,0 +1,22 @@
+package io.quarkus.it.kubernetes.client;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+@Path("/version")
+public class Version {
+
+    private final KubernetesClient kubernetesClient;
+
+    public Version(KubernetesClient kubernetesClient) {
+        this.kubernetesClient = kubernetesClient;
+    }
+
+    @GET
+    public Response version() {
+        return Response.ok(kubernetesClient.getKubernetesVersion().getPlatform()).build();
+    }
+}

--- a/integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientTest.java
+++ b/integration-tests/kubernetes-client/src/test/java/io/quarkus/it/kubernetes/client/KubernetesClientTest.java
@@ -48,6 +48,9 @@ public class KubernetesClientTest {
 
         RestAssured.when().post("/pod/test").then()
                 .body(containsString("54321"));
+
+        RestAssured.when().get("/version").then()
+                .statusCode(200);
     }
 
     @Test

